lib/core/include/irods/system_error.hpp

```diff
@@ -0,0 +1,80 @@
+#ifndef IRODS_SYSTEM_ERROR_HPP
+#define IRODS_SYSTEM_ERROR_HPP
+
+/// \file
+
+#include <compare>
+#include <system_error>
+
+namespace irods::experimental
+{
+    /// An error category that contains iRODS specific logic for dealing with
+    /// iRODS error codes.
+    ///
+    /// See https://cppreference.com for additional details about std::error_category.
+    ///
+    /// \since 4.2.12
+    class error_category : public std::error_category
+    {
+      public:
+        [[nodiscard]] auto name() const noexcept -> const char* override
+        {
+            return "iRODS";
+        } // name
+
+        [[nodiscard]] auto message(int _condition) const -> std::string override;
+
+        auto operator==(const error_category& _rhs) const noexcept -> bool
+        {
+            return this == &_rhs;
+        } // operator==
+
+        auto operator<=>(const error_category& _rhs) const noexcept -> std::strong_ordering
+        {
+            if (this < &_rhs) {
+                return std::strong_ordering::less;
+            }
+
+            if (this > &_rhs) {
+                return std::strong_ordering::greater;
+            }
+
+            return std::strong_ordering::equal;
+        } // operator<=>
+    }; // class error_category
+
+    /// Identifies the iRODS error category.
+    ///
+    /// Obtains a reference to the static error category object for iRODS errors. The object
+    /// is required to override the virtual function \p std::error_category::name() to return
+    /// a pointer to the string \p "iRODS". It is used to identify error conditions that
+    /// correspond to the iRODS error codes defined in rodsErrorTable.h.
+    ///
+    /// \return A reference to the static object derived from \p std::error_category.
+    ///
+    /// \since 4.2.12
+    [[nodiscard]] auto irods_category() noexcept -> const error_category&;
+
+    /// Creates a \p std::error_code object holding the value \p _ec.
+    ///
+    /// \param[in] _ec The integer to create the \p std::error_code object for.
+    ///
+    /// \return A \p std::error_code object.
+    ///
+    /// \since 4.2.6
```

Comment:
4.2.12?

Reply:
This function was part of the filesystem lib. It was always available, but usage was not advertised because it was seen as an implementation detail at the time.

With that said, do you feel it is more correct to still change it?

Reply:
I view this more as a removal and adding a new function (even though it is functionally identical), but I only pointed it out because I noticed that it was different. I'm fine with it either way

Reply:
After some discussion, we've landed on 4.2.12

Reply:
Done.